ergotree-interpreter/src/eval/sheader.rs

```diff
@@ -58,22 +61,31 @@ pub(crate) static HEIGHT_EVAL_FN: EvalFn = |_env, _ctx, obj, _args| {
 
 pub(crate) static MINER_PK_EVAL_FN: EvalFn = |_env, _ctx, obj, _args| {
     let header = obj.try_extract_into::<Header>()?;
-    Ok(header.miner_pk.into())
+    Ok(header.autolykos_solution.miner_pk.into())
 };
 
 pub(crate) static POW_ONETIME_PK_EVAL_FN: EvalFn = |_env, _ctx, obj, _args| {
     let header = obj.try_extract_into::<Header>()?;
-    Ok(header.pow_onetime_pk.into())
+    Ok(header
+        .autolykos_solution
+        .pow_onetime_pk
+        .unwrap_or_else(|| Box::new(EcPoint::default()))
+        .into())
 };
 
 pub(crate) static POW_NONCE_EVAL_FN: EvalFn = |_env, _ctx, obj, _args| {
     let header = obj.try_extract_into::<Header>()?;
-    Ok(header.nonce.into())
+    Ok(header.autolykos_solution.nonce.into())
 };
 
 pub(crate) static POW_DISTANCE_EVAL_FN: EvalFn = |_env, _ctx, obj, _args| {
     let header = obj.try_extract_into::<Header>()?;
-    let pow_distance: BigInt256 = header.pow_distance.try_into().map_err(EvalError::Misc)?;
+    let pow_distance: BigInt256 = header
+        .autolykos_solution
+        .pow_distance
+        .unwrap_or_default()
```

Comment:
Assuming a `pow_distance` of zero, if not specified (if autolykos is v2)

Reply:
Yes. Thank you for bringing my attention to these details! I appreciate it.